### PR TITLE
Add initial docs style guide

### PR DESCRIPTION
**Pull request by dfarrell07**
Start a documentation style guide so the community can agree on and
follow best practices as they contribute docs, especially during the
upcoming docs-only sprint.

Follow the upstream K8s Docs Style Guide as a base.

This guide should be extended as contribution/review shows gaps.

Closes: #100

Co-authored-by: Thomas Pantelis <tompantelis@gmail.com>
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

File: src/content/contributing/website/style_guide.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+This guide is meant to help keep our documentation consistent and ease the
+contribution and review process.
+
+### Submariner project and binary capitalization
+
+The project name "Submariner" should be capitalized. The same goes for
+Submariner subprojects like Lighthouse, Shipyard, Coastguard and similar.
+
+To indicate that a Submariner project artifact should taken as the literal
```
Comment (mangelajo) [inline src/content/contributing/website/style_guide.md:17]:
```suggestion
To indicate that a Submariner project artifact should be taken as the literal
```
?

Comment (dfarrell07) [reply src/content/contributing/website/style_guide.md]: Yes, good catch thanks!

File: src/content/contributing/website/style_guide.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+### Submariner project and binary capitalization
+
+The project name "Submariner" should be capitalized. The same goes for
+Submariner subprojects like Lighthouse, Shipyard, Coastguard and similar.
+
+To indicate that a Submariner project artifact should be taken as the literal
+characters of a command, Submariner project binaries like `subctl` can be
+written with \`backticks\`.
+
```
Comment (sridhargaddam) [inline src/content/contributing/website/style_guide.md:20]: I think you can include a section with a note on K8s objects (like Pod/Service/Deployment etc) as we use them frequently in our documentation.

Comment (dfarrell07) [reply src/content/contributing/website/style_guide.md:20]: Nice suggestion @sridhargaddam, thanks! While starting to add this I was digging around and noticed that the K8s style guide covers it nicely in their docs about object capitalization. Maybe I can re-work this PR to first say we follow the K8s Style Guide (which would cover this rule), and extend parts of it (like their [word list](https://kubernetes.io/docs/contribute/style/style-guide/#kubernetes-io-word-list)) to add Submariner-specific things.

Comment (dfarrell07) [reply src/content/contributing/website/style_guide.md:19]: Done in the latest. :)

File: src/content/contributing/website/style_guide.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+A list of Submariner-specific terms and words to be used consistently across
+the site.
+
+Term | Usage
+:--- | :----
+Submariner | The project name *Submariner* should always be capitalized.
+Admral | The project name *Admral* should always be capitalized.
+Lighthouse | The project name *Lighthouse* should always be capitalized.
+Coastguard | The project name *Coastguard* should always be capitalized.
+`subctl` | The artfiact `subctl` should not be capitalized and should be formated in code style.
```
Comment (tpantelis) [inline src/content/contributing/website/style_guide.md:26]:
```suggestion
`subctl` | The artifact `subctl` should not be capitalized and should be formatted in code style.
```

File: src/content/contributing/website/style_guide.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+A list of Submariner-specific terms and words to be used consistently across
+the site.
+
+Term | Usage
+:--- | :----
+Submariner | The project name *Submariner* should always be capitalized.
+Admral | The project name *Admral* should always be capitalized.
+Lighthouse | The project name *Lighthouse* should always be capitalized.
+Coastguard | The project name *Coastguard* should always be capitalized.
```
Comment (tpantelis) [inline src/content/contributing/website/style_guide.md:25]: It seems we should we mention Shipyard as well.

Comment (dfarrell07) [reply src/content/contributing/website/style_guide.md:25]: Good catch, done in the latest.

File: src/content/contributing/website/style_guide.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+### Submariner.io word list
+
+A list of Submariner-specific terms and words to be used consistently across
+the site.
+
+Term | Usage
+:--- | :----
+Submariner | The project name *Submariner* should always be capitalized.
+Admral | The project name *Admral* should always be capitalized.
```
Comment (sridhargaddam) [inline src/content/contributing/website/style_guide.md:23]: nit: Admiral?

File: src/content/contributing/website/style_guide.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+Lighthouse | The project name *Lighthouse* should always be capitalized.
+Coastguard | The project name *Coastguard* should always be capitalized.
+Shipyard | The project name *Shipyard* should always be capitalized.
+`subctl` | The artifact `subctl` should not be capitalized and should be formatted in code style.
```
Comment (tpantelis) [inline src/content/contributing/website/style_guide.md:27]: Should we extend this to any script or command?

Comment (dfarrell07) [reply src/content/contributing/website/style_guide.md:27]: Maybe, I'm not sure. I think this is partly covered by https://kubernetes.io/docs/contribute/style/style-guide/#use-code-style-for-kubernetes-command-tool-and-component-names, and I just intended to be explicit with major/common Submariner terms.

File: src/content/contributing/website/style_guide.md (hunk 5 above)
Comment (dfarrell07) [reply src/content/contributing/website/style_guide.md]: Ah, thanks, fixed.

**Pull request merged**